Azure Identity lib docs updates (#45566)
Azure/azure-sdk-for-net: sdk/identity/Azure.Identity/README.md

### Define a custom authentication flow with `ChainedTokenCredential`

> While `DefaultAzureCredential` is generally the quickest way to authenticate apps for Azure, you can customize the credentials that are considered. `ChainedTokenCredential` enables users to combine multiple credential instances to define a customized chain of credentials. For more information, see [ChainedTokenCredential overview][ctc_overview].

Reviewer: When I initially read the first sentence it sounded like DAC could be customized with custom credentials.

```suggestion
While `DefaultAzureCredential` is generally the quickest way to authenticate apps for Azure, you can create a customized chain of credentials to be considered. `ChainedTokenCredential` enables users to combine multiple credential instances to define a customized chain of credentials. For more information, see [ChainedTokenCredential overview][ctc_overview].
```

var client = new SecretClient(new Uri("https://myvault.vault.azure.net/"), credential);
```

## Sovereign cloud configuration

> By default, credentials authenticate to the Microsoft Entra endpoint for the Azure Public Cloud. To access resources in other clouds, such as Azure US Government or a private cloud, configure credentials with the `AuthorityHost` argument. [AzureAuthorityHosts](https://learn.microsoft.com/dotnet/api/azure.identity.azureauthorityhosts?view=azure-dotnet) defines authorities for well-known clouds:

Reviewer: [AuthorityHost](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/TokenCredentialOptions.cs#L28) is set on `TokenCredentialOptions`, so the term option seems more natural here.

```suggestion
By default, credentials authenticate to the Microsoft Entra endpoint for the Azure Public Cloud. To access resources in other clouds, such as Azure US Government or a private cloud, configure credentials with the `AuthorityHost` option. [AzureAuthorityHosts](https://learn.microsoft.com/dotnet/api/azure.identity.azureauthorityhosts?view=azure-dotnet) defines authorities for well-known clouds:
```
